### PR TITLE
[wip] perf(core): optimize async ops and event loop

### PR DESCRIPTION
**Pull request by littledivy**
WIP.

### CODE REVIEW

File: core/runtime.rs [hunk 1]
```diff
@@ -1882,100 +1922,40 @@ impl JsRuntime {
...
+    for (realm, resolver, mut resp) in results {
+      let scope = &mut realm.handle_scope(isolate);
+      let resolver = resolver.open(scope);
+      let result = match resp.to_v8(scope) {
+        Ok(v) => v,
+        Err(e) => OpResult::Err(OpError::new(&|_| "TypeError", e.into()))
+          .to_v8(scope)
+          .unwrap(),
+      };
+      resolver.resolve(scope, result);
```
Comment (aapoalas) [inline core/runtime.rs:1957]: Use reject on error?

Comment (nayeemrmn) [reply core/runtime.rs:1957]: ^ We have `crate::error::to_v8_error(scope, op_state.get_error_class_fn, &e)`

File: ops/lib.rs [hunk 2]
```diff
@@ -347,172 +341,166 @@ fn codegen_fast_impl(
...
-                Default::default()
-              },
+    let (trampoline, raw_block) = if is_async {
+      (
+        quote! {
+          fn #func_name (_recv: #core::v8::Local<#core::v8::Object>, __promise_id: i32, __promise_resolver: v8::Local<v8::Function>, #(#inputs),*) -> i32 {
+            #recv_decl
+            let __op_id = ctx.id;
+            let isolate = unsafe { &mut *ctx.isolate };
+            let resolver = #core::v8::Global::<#core::v8::Function>::new(isolate, __promise_resolver);
```
Comment (aapoalas) [inline ops/lib.rs:424]: Does this not assert? Holy moly, I would've 100% assumed creating globals would be forbidden by Fast API asserts.

Comment (aapoalas) [reply ops/lib.rs:424]: From V8 dev: Creating should be Global is fine (although perhaps not quite what is expected). cc @MayaLekova

**Pull request closed without merging**